### PR TITLE
xenstore: add xs access routines

### PR DESCRIPTION
**Pull request by sa-kib**
Initial code for read & write to XS, also stub for permission setting.

### CODE REVIEW

File: xenstore-srv/include/xss.h [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#pragma once
```
Comment (firscity) [inline xenstore-srv/include/xss.h:1]: Please add SPDX identifier and EPAM copyright.

File: xenstore-srv/include/xss.h [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+#include <zephyr/xen/public/xen.h>
+
+enum xs_perm {
+	XS_PERM_NONE = 0x0,
+	XS_PERM_READ = 0x1,
+	XS_PERM_WRITE = 0x2,
+	XS_PERM_BOTH = XS_PERM_WRITE | XS_PERM_READ
+};
+
+int xss_read(const char *path, char *value, size_t len);
```
Comment (firscity) [inline xenstore-srv/include/xss.h:11]: Please add functions description

File: xenstore-srv/src/xenstore_srv.c [hunk 3]
```diff
@@ -317,6 +318,39 @@ void xss_do_write(char *const_path, char *data)
...
+int xss_write(const char *path, const char *value)
+{
+	xss_do_write(path, value);
+	return 0;
+}
+
+int xss_read(const char *path, char *value, size_t len)
+{
+	int rc = -ENOENT;
+	k_mutex_lock(&xsel_mutex, K_FOREVER);
```
Comment (firscity) [inline xenstore-srv/src/xenstore_srv.c:330]: Please, add space after declaration.

File: xenstore-srv/src/xenstore_srv.c [hunk 4]
```diff
@@ -317,6 +318,39 @@ void xss_do_write(char *const_path, char *data)
...
+	}
+	k_mutex_unlock(&xsel_mutex);
+	return rc;
+}
+
+int xss_read_integer(const char *path, int *value)
+{
+	char ns[32] = { 0 };
+	int rc = xss_read(path, ns, sizeof(ns));
+	if (!rc)
```
Comment (firscity) [inline xenstore-srv/src/xenstore_srv.c:344]: Please, add space after declaration.

**Review by firscity — CHANGES_REQUESTED** [review]
Please, also add why we need it in commit message.

File: xenstore-srv/include/xss.h (hunk 1 above)
Comment (sa-kib) [reply xenstore-srv/include/xss.h:7]: added in 2e0617e

File: xenstore-srv/include/xss.h (hunk 2 above)
Comment (sa-kib) [reply xenstore-srv/include/xss.h:25]: added in 2e0617e

File: xenstore-srv/src/xenstore_srv.c (hunk 3 above)
Comment (sa-kib) [reply xenstore-srv/src/xenstore_srv.c:331]: added in 2e0617e

File: xenstore-srv/src/xenstore_srv.c (hunk 4 above)
Comment (sa-kib) [reply xenstore-srv/src/xenstore_srv.c:349]: added in 2e0617e

**Comment by sa-kib** [comment]
> Please, also add why we need it in commit message.

commit msg updated

**Review by firscity — APPROVED** [review]
Please, add RB and force push squashed commit
`Reviewed-by: Dmytro Firsov <dmytro_firsov@epam.com>`

**Comment by sa-kib** [comment]
squashed & rebased

**Pull request merged**